internal/configtxgen/genesisconfig/config.go

```diff
@@ -221,17 +226,42 @@ func LoadTopLevel(configPaths ...string) *TopLevel {
 	}
 	logger.Debugf("Using config file: %s", config.ConfigFileUsed())
 
-	var uconf TopLevel
-	err = viperutil.EnhancedExactUnmarshal(config, &uconf)
+	uconf, err := loadConfigCache(config, config.ConfigFileUsed())
 	if err != nil {
-		logger.Panic("Error unmarshaling config into struct: ", err)
+		logger.Panic("failed to load configCache: %s", err)
 	}
 
-	(&uconf).completeInitialization(filepath.Dir(config.ConfigFileUsed()))
+	uconf.completeInitialization(filepath.Dir(config.ConfigFileUsed()))
 
 	logger.Infof("Loaded configuration: %s", config.ConfigFileUsed())
 
-	return &uconf
+	return uconf
+}
+
+// loadConfigCache loads the TopLevel config structure from configCache.
+// if not successful, it unmarshal a config file, and populate configCache
+// with marshaled TopLevel struct.
+func loadConfigCache(config *viper.Viper, configPath string) (*TopLevel, error) {
+	conf := &TopLevel{}
+	serializedConf, ok := configCache[configPath]
+	if !ok {
+		err := viperutil.EnhancedExactUnmarshal(config, conf)
+		if err != nil {
+			logger.Panic("Error unmarshaling config into struct: ", err)
```

Comment:
This function returns an error; the caller will panic when it receives the error. Why is this panicking?

Reply:
done